app/controllers/api/apps_controller.rb

```diff
@@ -46,4 +48,15 @@ def increase_log_count
       log_count: @app.log_count,
     }
   end
+
+  def authorize
+    @current_user = User.find_by_username_or_email(params[:username])
+    @infrastructure = Infrastructure.find_by_cluster_name(params[:cluster_name])
+    unless InfrastructurePolicy.new(@current_user, @infrastructure).exist?
+      render json: "Unauthorized", status: :unauthorized
```

Comment:
mas @pushm0v, I think we can refactor this code and move it to `Api::BaseController` so it will look like what we have in [`ApplicationController`](https://github.com/BaritoLog/BaritoMarket/blob/master/app/controllers/application_controller.rb#L6). And this code can be replaced with only `authorize @infrastructure, :exists?`.

Reply:
got it, will refactor it... 👍